azurerm/internal/services/managedapplication/parse/managed_application_definition_test.go

```diff
@@ -0,0 +1,81 @@
+package parse
+
+import (
+	"testing"
+)
+
+func TestManagedApplicationDefinitionID(t *testing.T) {
+	testData := []struct {
+		input    string
+		expected *ManagedApplicationDefinitionId
+	}{
+		{
+			input:    "",
+			expected: nil,
+		},
+		{
+			input:    "/subscriptions/00000000-0000-0000-0000-000000000000",
+			expected: nil,
+		},
+		{
+			input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups",
+			expected: nil,
+		},
+		{
+			input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/hello",
+			expected: nil,
+		},
+		{
+			input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/hello/providers",
+			expected: nil,
+		},
+		{
+			input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/hello/providers/Microsoft.Solutions",
+			expected: nil,
+		},
+		{
+			input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/hello/providers/Microsoft.Solutions/applicationDefinitions",
+			expected: nil,
+		},
+		{
+			input: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/hello/providers/Microsoft.Solutions/applicationDefinitions/appDef1",
+			expected: &ManagedApplicationDefinitionId{
+				Name:          "appDef1",
+				ResourceGroup: "hello",
+			},
+		},
+	}
+
+	for _, v := range testData {
+		t.Logf("[DEBUG] Testing %q..", v.input)
+		actual, err := ManagedApplicationDefinitionID(v.input)
+
+		// if we get something there shouldn't be an error
+		if v.expected != nil && err == nil {
```

Comment:
I maybe dumb here, but if we expect something and there is no error, why do we just continue instead of checking anything? Maybe you can take some other UT of parse function as a reference. My personally prefer to explicitly define another field in `testData` to indicate whether we expect an error.

Reply:
updated